### PR TITLE
tl: fix compiler version checks

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **tl/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

I'm trying to fix recipes which doesn't behave as expected while checking min compiler version (too restrictive) in `configure()`

### CODE REVIEW

File: recipes/tl/all/conanfile.py [hunk 1]
```diff
@@ -26,18 +29,22 @@ def configure(self):
             "Visual Studio": "15"
         }
 
+        def lazy_lt_compiler_version(version1, version2):
+            v1, *v1_res = version1.split(".", 1)
+            v2, *v2_res = version2.split(".", 1)
+            if v1 == v2 and v1_res and v2_res:
+                return lazy_lt_compiler_version(v1_res, v2_res)
+            return v1 < v2
```
Comment (madebr) [inline recipes/tl/all/conanfile.py:37]: Can this be replaced with the following
```python
version1.split(".") < version2.split(".")
```

Comment (SpaceIm) [reply recipes/tl/all/conanfile.py:37]: No:

```
>>> "1.7".split(".") < "1.7.3".split(".")
True
```

Comment (madebr) [reply recipes/tl/all/conanfile.py:37]: Indeed.

How about this one?

```python
def lazy_lt(v1, v2):
  lv1 = v1.split(".")
  lv2 = v2.split(".")
  lv1 = lv1 + ["0"] * (max((len(lv1), len(lv2))) - len(lv1))
  lv2 = lv2 + ["0"] * (max((len(lv1), len(lv2))) - len(lv2))
  return lv1 < lv2
```

Comment (SpaceIm) [reply recipes/tl/all/conanfile.py:37]: Probably not.

But this works:

```python
def lazy_lt_version(v1, v2):
    lv1 = v1.split(".")
    lv2 = v2.split(".")
    min_length = min(len(lv1), len(lv2))
    return lv1[:min_length] < lv2[:min_length]
```

Comment (madebr) [reply recipes/tl/all/conanfile.py:37]: Yours does not compute `lazy_lt_version("1.2", "1.2.3")` correctly.

Comment (SpaceIm) [reply recipes/tl/all/conanfile.py]: I want 1.2 < 1.2.3 to be False

Comment (madebr) [reply recipes/tl/all/conanfile.py]: How about `lazy_lt_version("1.2.3", "1.2")` then?

Comment (SpaceIm) [reply recipes/tl/all/conanfile.py]: False also

Comment (madebr) [reply recipes/tl/all/conanfile.py]: Ok then. I guess I don't fully understand your intention.

**Review by madebr — APPROVED** [review]

File: recipes/tl/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/tl/all/conanfile.py]: My intention is to not raise if compiler.version is 6 for gcc, because min version is gcc 6.4

Comment (madebr) [reply recipes/tl/all/conanfile.py]: Ooh, that makes total sense.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**